ducktape/tests/runner.py

```diff
@@ -195,6 +203,14 @@ def teardown_single_test(self, teardown_services=True):
             exceptions.append(e)
             self.log(logging.WARN, "Error freeing nodes: %s" % e.message + "\n" + traceback.format_exc(limit=16))
 
+        # Remove reference to services. This is important to prevent potential memory leaks if users write services
+        # which themselves have references to large memory-intensive objects
+        try:
+            del self.current_test_context.services
+        except BaseException as e:
```

Comment:
What are we expecting to catch here? Lack of `services` field or something? Seems odd that a `del` like this would be a problem.